### ISSUE TITLE
KL bias correction with importance_sampling_level="sequence" broadcasts a sequence-level ratio onto per-token KL

### ISSUE THREAD

**Issue by gowtham-sai-yadav** [huggingface/trl#6586]
### Description

Follow-up to #6503, splitting out a caveat @albertvillanova raised in review there.

`use_bias_correction_kl` multiplies the per-token KL by `coef_1`, but `coef_1`'s meaning depends on `importance_sampling_level`:

```python
# trl/trainer/grpo_trainer.py
log_ratio = per_token_logps - old_per_token_logps
if self.importance_sampling_level == "token":
    log_importance_weights = log_ratio                                    # (B, T)
elif self.importance_sampling_level == "sequence":
    log_importance_weights = (log_ratio * mask).sum(-1) / mask.sum(-1).clamp(min=1.0)
    log_importance_weights = log_importance_weights.unsqueeze(-1)         # (B, 1)

coef_1 = torch.exp(log_importance_weights)
...
if self.args.use_bias_correction_kl:
    per_token_kl = per_token_kl * coef_1
```

With `importance_sampling_level="token"` this is the per-token correction from the [DeepSeek-V3.2 paper](https://huggingface.co/papers/2512.02556), and the KL gradient is the unbiased reverse-KL gradient.

With `"sequence"`, `coef_1` is the sequence-mean ratio of shape `(B, 1)`, so a single sequence-level weight is broadcast across the per-token KL. The gradient then picks up an extra `(Σ_t k3_t) · ∇(mean_t log π_t)` term, so it is neither the paper's per-token correction nor the unbiased reverse-KL gradient in that configuration.

This is pre-existing behaviour and not a crash. It matters a bit more now that #6503 made the correction the default, so a user combining it with `importance_sampling_level="sequence"` (for example the GSPO setup) silently gets a different KL gradient than the docstring describes. The docstring added in #6503 currently qualifies the claim to `"token"`, which is why this is a follow-up rather than part of that PR.

### Possible directions

1. Use the per-token ratio for the KL term regardless of `importance_sampling_level`, so the correction always matches the paper.
2. Keep the current behaviour and document it in more detail as a known limitation of the sequence level.
3. Warn when `use_bias_correction_kl=True` is combined with `importance_sampling_level="sequence"`.

I don't have a strong view on which is right, since option 1 changes behaviour for existing sequence-level runs. Happy to open a PR once you decide the direction.
